### PR TITLE
Add code spellcheck CI job

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

Fix spelling mistakes and create codespell CI job.
An extension of https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3422 with the remaining mistakes fixed

### Backward compatibility

Yes

### CODE REVIEW

File: .codespell/.codespellrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[codespell]
+skip = .git,node_modules,yarn.lock,Cargo.lock,./typescript/helloworld
```
Comment (jmrossy) [inline .codespell/.codespellrc:2]: There's some bug with codespell causing it to trip on `receivedFrom` in helloworld even though it's on the ignore list 🤷 
Other than that one occurrence it works fine.

File: rust/hyperlane-core/src/traits/validator_announce.rs [hunk 2]
```diff
@@ -24,6 +24,6 @@ pub trait ValidatorAnnounce: HyperlaneContract + Send + Sync + Debug {
     ) -> ChainResult<TxOutcome>;
 
     /// Returns the number of additional tokens needed to pay for the announce
-    /// transaction. Return `None` if the needed tokens canno tbe determined.
+    /// transaction. Return `None` if the needed tokens canno the determined.
```
Comment (paulbalaji) [inline rust/hyperlane-core/src/traits/validator_announce.rs:27]:
```suggestion
    /// transaction. Return `None` if the needed tokens cannot be determined.
```

File: typescript/infra/src/utils/metrics.ts [hunk 3]
```diff
@@ -47,7 +47,7 @@ export async function submitMetrics(
  * Start a simple HTTP server to host metrics. This just takes the registry and dumps the text
  * string to people who request `GET /metrics`.
  *
- * PROMETHEUS_PORT env var is used to determine what port ot host on, defaults to 9090.
+ * PROMETHEUS_PORT env var is used to determine what host port, defaults to 9090.
```
Comment (paulbalaji) [inline typescript/infra/src/utils/metrics.ts:50]:
```suggestion
 * PROMETHEUS_PORT env var is used to determine what port to host on, defaults to 9090.
```

**Review by paulbalaji — APPROVED** [review]
couple nits

**Review by yorhodes — APPROVED** [review]

**Pull request merged**